### PR TITLE
Query: Rewrite entity equality for dependents sharing table

### LINKED ISSUE #20342: Entity Equality rewrites owned == null check to PK check
Even if owned entity is null, the PK may have value in case of table splitting scenario.

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #20342

### CODE REVIEW

File: src/EFCore.Relational/Query/RelationalSqlTranslatingExpressionVisitor.cs [hunk 1]
```diff
@@ -1171,11 +1210,15 @@ private bool TryRewriteEntityEquality(ExpressionType nodeType, Expression left,
             }
 
             result = Visit(primaryKeyProperties.Select(p =>
-                    Expression.MakeBinary(
-                        nodeType,
-                        CreatePropertyAccessExpression(left, p),
-                        CreatePropertyAccessExpression(right, p)))
-                    .Aggregate((l, r) => Expression.AndAlso(l, r)));
+                {
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/RelationalSqlTranslatingExpressionVisitor.cs:1213]: This extends the fix for #19407 for more scenarios.

File: src/EFCore/Query/Internal/NavigationExpandingExpressionVisitor.ExpressionVisitors.cs [hunk 2]
```diff
@@ -459,7 +459,7 @@ private Expression TryExpandNavigation(Expression root, MemberIdentity memberIde
                 return innerSource.PendingSelector;
             }
 
-            static Expression AddConvertToObject(Expression expression)
+            private static Expression AddConvertToObject(Expression expression)
```
Comment (maumar) [inline src/EFCore/Query/Internal/NavigationExpandingExpressionVisitor.ExpressionVisitors.cs:462]: convert it into local function?

Comment (smitpatel) [reply src/EFCore/Query/Internal/NavigationExpandingExpressionVisitor.ExpressionVisitors.cs:462]: I thought about it. Initially I intended to write it as a local function and I thought I did. But then VS said, I need to add accessibility modifier and added private in front of it. So I realized it is not local function.
But then
Inside an .Internal class
in a nested clas
in a private method
Does it really matter ? :trollface:

**Pull request merged**